controllers/sandbox_controller.go

```diff
@@ -70,58 +75,119 @@ func (r *SandboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
-	// Check if the pod already exists, if not create a new one
-	podInCluster := &corev1.Pod{}
-	found := true
-	err := r.Get(ctx, types.NamespacedName{Name: sandbox.Name, Namespace: sandbox.Namespace}, podInCluster)
+	_, err := r.reconcilePod(ctx, sandbox)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	_, err = r.reconcileService(ctx, sandbox)
 	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Update the sandbox status
+	// TODO: Update the sandbox status
+
+	return ctrl.Result{}, nil
+}
+
+func (r *SandboxReconciler) reconcileService(ctx context.Context, sandbox *sandboxv1alpha1.Sandbox) (*corev1.Service, error) {
+	log := log.FromContext(ctx)
+	service := &corev1.Service{}
+	found := true
```

Comment:
nit: default `found` to false and only set to true when it's found. This avoids mistakenly assuming resources are found when more logic is added to the code block below.

Reply:
done.